card/mod/core/set/all/fetch.rb

```diff
@@ -124,23 +124,29 @@ def fetch opts={}
   end
 end
 
-def renew args={}
-  opts = args[:new].clone
+def renew new_opts
+  return self if new_opts.blank?
+  opts = new_opts.clone
+  copy = dup
```

Comment:
the main change to fetch here is that renew makes a duplicate rather than instantiating a new object.  That way cached attributes are retained.